nebula-logger/core/main/logger-engine/classes/LogEntryEventBuilder.cls

```diff
@@ -483,6 +489,40 @@ global with sharing class LogEntryEventBuilder {
         return this;
     }
 
+    /**
+     * @description Sets the log entry event's HTTP Request fields
+     * @param  request The instance of `HttpRequest` to log
+     * @return              The same instance of `LogEntryEventBuilder`, useful for chaining methods
+     */
+    global LogEntryEventBuilder setHttpRequestDetails(HttpRequest request) {
+        if (this.shouldSave == false || request == null) {
+            return this;
+        }
+
+        this.logEntryEvent.HttpRequestBody__c = request.getBody();
+        this.logEntryEvent.HttpRequestCompressed__c = request.getCompressed();
+        this.logEntryEvent.HttpRequestEndpoint__c = request.getEndpoint();
+        this.logEntryEvent.HttpRequestMethod__c = request.getMethod();
+        return this;
+    }
+
+    /**
+     * @description Sets the log entry event's HTTP Response fields
+     * @param  response The instance of `HttpResponse` to log
+     * @return              The same instance of `LogEntryEventBuilder`, useful for chaining methods
+     */
+    global LogEntryEventBuilder setHttpResponseDetails(HttpResponse response) {
+        if (this.shouldSave == false || response == null) {
+            return this;
+        }
+
+        this.logEntryEvent.HttpResponseBody__c = response.getBody();
```

Comment:
I'm assuming this gets sanitized downstream(?), but leaving this question here in the event that I don't see/miss that!

Reply:
Oooh, no, it doesn't, that's a great call - I think I'll run it through the `applyDataMaskRules()` method to obfuscate any sensitive data, but did you have anything else in mind?

Reply:
nope, that's all I was thinking of!

Reply:
I just realized there's the same issue for `HttpRequestBody__c` - I'll run that field through the data mask rules too.

Reply:
Oof, nice catch!